test/multiverse/suites/roda/roda_instrumentation_test.rb

```diff
@@ -0,0 +1,155 @@
+      get('/home')
+      txn = harvest_transaction_events![1][0]
+
+      assert_equal 'Controller/Roda/RodaTestApp/home', txn[0]['name']
+      assert_equal 200, txn[2][:'http.statusCode']
+    end
+  end
+
+  def test_http_verb_request_no_request_method
+    fake_request = Struct.new('FakeRequest', :path).new
+    name = NewRelic::Agent::Instrumentation::Roda::TransactionNamer.transaction_name(fake_request)
+
+    assert_equal ::NewRelic::Agent::UNKNOWN_METRIC, name
+  end
+
+  def test_request_is_recorded
+    get('/home')
+    txn = harvest_transaction_events![1][0]
+
+    assert_equal 'Controller/Roda/RodaTestApp/GET home', txn[0]['name']
+    assert_equal 200, txn[2][:'http.statusCode']
+  end
+
+  def test_500_response_status
+    get('/error')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events!
+
+    assert_equal 500, txn[1][0][2][:"http.statusCode"]
+    assert_equal 'Oh No!', last_response.body
+    assert_equal 1, errors.size
+  end
+
+  def test_404_response_status
+    get('/nothing')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events!
+
+    assert_equal 404, txn[1][0][2][:"http.statusCode"]
+    assert_equal 0, errors.size
+  end
+
+  def test_empty_route_name_and_response_status
+    get('')
+    errors = harvest_error_traces!
+    txn = harvest_transaction_events![1][0]
+
+    assert_equal 'Controller/Roda/RodaTestApp/GET /', txn[0]['name']
+    assert_equal 302, txn[2][:'http.statusCode']
+  end
+
+  def test_roda_auto_middleware_disabled
+    with_config(:disable_roda_auto_middleware => true) do
+      RodaNoMiddleware.build_rack_app_with_tracing {}
+
+      assert_truthy NewRelic::Agent::Agent::config[:disable_roda_auto_middleware]
+    end
+  end
+
+  def test_roda_instrumentation_works_if_middleware_disabled
+    with_config(:disable_middleware_instrumentation => true) do
+      get('/home')
+      txn = harvest_transaction_events![1][0]
+
+      assert_equal 'Controller/Roda/RodaTestApp/GET home', txn[0]['name']
+    end
+  end
+
+  def test_transaction_name_error
+    NewRelic::Agent.stub(:logger, NewRelic::Agent::MemoryLogger.new) do
+      # pass in {} to produce an error, because {} doesn't support #path and
+      # confirm that the desired error handling took place
+      result = NewRelic::Agent::Instrumentation::Roda::TransactionNamer.transaction_name({})
+
+      assert_equal NewRelic::Agent::UNKNOWN_METRIC, result
+      assert_logged(/NoMethodError.*Error encountered trying to identify Roda transaction name/)
+    end
+  end
+
+  def test_rack_request_params_error
+    NewRelic::Agent.stub(:logger, NewRelic::Agent::MemoryLogger.new) do
+      begin
+        RodaTestApp::RodaRequest.any_instance
+          .stubs(:params).raises(StandardError.new)
+        get('/home?')
+      rescue
+        # NOOP - Allow error to be raised
+      ensure
+        assert_logged(/Failed to get params from Rack request./)
+      end
+    end
```

Comment:
Same deal here on not needing to rescue or noop.

```suggestion
    NewRelic::Agent.stub(:logger, NewRelic::Agent::MemoryLogger.new) do
      # Have the #params call made on the request raise an exception to test
      # the error handling
      RodaTestApp::RodaRequest.any_instance.stubs(:params).raises(StandardError.new)
      get('/home?')

      assert_logged(/Failed to get params from Rack request./)
    end
```

Reply:
THank you—great suggestion and makes sense. Committed!